redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -218,6 +212,19 @@ private class LazyListContainerCell(frame: CValue<CGRect>) : UICollectionViewCel
     super.layoutSubviews()
     widgetView?.setFrame(this.contentView.bounds)
   }
+
+  override fun preferredLayoutAttributesFittingAttributes(
+    layoutAttributes: UICollectionViewLayoutAttributes,
+  ): UICollectionViewLayoutAttributes {
+    val itemSize = widgetView!!.sizeThatFits(collectionView.frame.useContents { size.readValue() })
+    return layoutAttributes.apply {
+      size = if (collectionViewFlowLayout.scrollDirection == UICollectionViewScrollDirectionVertical) {
```

Comment:
Have you tried setting `frame.size` instead of `size`? Not sure if that would impact performance one way or the other.

Reply:
I have but it just didn't apply any constraints. It just defaults to `itemSize`, [which is 50x50 by default](https://developer.apple.com/documentation/uikit/uicollectionviewflowlayout/1617711-itemsize), so ends up appearing in a grid with each item clipped.